### ISSUE TITLE
Manage "sparse" attribute option for Index MongoDB

### ISSUE THREAD

**Issue by TylerR909** [strapi/strapi#6297]
**Describe the bug**
My model has a field which must be unique but can also be nullable. To accomplish this I did some research and discovered I can add `sparse: true` to my model's field like so:

```
    "myField": {
      "type": "string",
      "unique": true,
      "sparse": true
    },
```

However, back in the UI I'm trying to extend my model with new fields and getting a generic "An error occurred" message. Digging into the Network request and Console reveals this:

![image](https://user-images.githubusercontent.com/20718430/82461808-a942dc00-9a6f-11ea-9388-b75028f86610.png)

**Steps to reproduce the behavior**

1. Add `sparse: true` to one of your models
2. In the UI, try to update that model by adding new fields
3. Witness generic "An error occurred" message
4. Open console to reveal error

**Expected behavior**
Sparse/nullable fields don't break model updates in the UI. It would also be nice, when creating a new field in the UI, to be able to select Unique=True and Nullable=True in Advanced. Or infer `sparse: true` if Unique=True and Required=False.

**Current Workaround**
Temporarily remove `sparse: true` from my model, make my updates (Was adding a component and wanted to let the UI do the heavy-lifting), add back `sparse: true` where desired.

**System**

- Node.js version: v10.16.3
- NPM version: 6.9.0
- Strapi version: 3.0.0-beta.20.2
- Database: MongoDB
- Operating system: OSX Mojave 10.14.6

**Comment by derrickmehaffy** [comment]
I don't believe that `sparse` is an option? Where did you find this?

**Comment by TylerR909** [comment]
Maybe it's a MongoDB-only thing? I had to migrate over from SQLite to MongoDB a couple days ago and all my saves stopped working because `unique` was failing to save "duplicate" `null` values.

A couple google searches later, I discovered sparse, gave it a shot, and it started working.

Got tons of hits on stuff like [this SO question](https://stackoverflow.com/questions/40759941/unique-if-not-null-check-in-mongoose). Tried it and it worked immediately. 

If you have an alternate way of verifying "Not required. Unique if not null" I'm open to ideas. But this was a pretty straightforward fix for me until it randomly broke the UI.
